routing_algorithms/shortest_path.hpp

```diff
@@ -148,39 +150,69 @@ class ShortestPathRouting final
                 // SimpleLogger().Write(logDEBUG) << "rev-a insert: " <<
                 // phantom_node_pair.target_phantom.reverse_node_id << ", w: " << phantom_node_pair.target_phantom.GetReverseWeightPlusOffset();
             }
+			//TIMER_STOP( t_init );
+			//std::cout << "[init] " << TIMER_MSEC( t_init ) << std::endl;
 
+			//ch stopping criterion
+			int local_minimum_forward = 0;
+			int local_minimum_reverse = 0;
+
+			//TIMER_START( t_query_1 );
             // run two-Target Dijkstra routing step.
-            while (0 < (forward_heap1.Size() + reverse_heap1.Size()))
-            {
+            while (0 < (forward_heap1.Size() + reverse_heap1.Size()) && local_upper_bound1 > local_minimum_forward + local_minimum_reverse )
```

Comment:
This termination criterion is only correct when running on the un-contracted core. Remove.